modules/percolator/src/main/java/org/elasticsearch/percolator/PercolatorFieldMapper.java

```diff
@@ -566,6 +573,90 @@ static void verifyQuery(QueryBuilder queryBuilder) {
         }
     }
 
+    static QueryBuilder rewrite(QueryBuilder queryBuilder) {
```

Comment:
I don't like this very much to be honest, I do wonder if we can introduce a special rewrite context that we can check for in the RangeQuery and then rewrite to `MatchAllDocs` this would make this change more contained?

Reply:
> I do wonder if we can introduce a special rewrite context that we can check for in the RangeQuery and then rewrite to MatchAllDocs

👍 That sounds like a good solution.

Reply:
@s1monw @jimczi I've pushed this: 21b7b869d8ac14ab266ea13ed69dd6eee8683c0b